osdk/tests/examples_in_book/write_a_kernel_in_100_lines_templates/lib.rs

```diff
@@ -57,29 +65,29 @@ fn create_user_space(program: &[u8]) -> UserSpace {
             cursor.map(frame.into(), map_prop);
         }
         drop(cursor);
-        Arc::new(vm_space)
+        vm_space
     };
-    let user_cpu_state = {
+    let user_ctx = {
         const ENTRY_POINT: Vaddr = 0x0040_1000; // The entry point for statically-linked executable
 
         // The user-space CPU states can be initialized
-        // to arbitrary values via the UserContext
+        // to arbitrary values via the `UserContext`
         // abstraction.
-        let mut user_cpu_state = UserContext::default();
-        user_cpu_state.set_rip(ENTRY_POINT);
-        user_cpu_state
+        let mut user_ctx = UserContext::default();
+        user_ctx.set_rip(ENTRY_POINT);
+        user_ctx
     };
-    UserSpace::new(user_address_space, user_cpu_state)
+    (user_space, user_ctx)
 }
 
-fn create_user_task(user_space: Arc<UserSpace>) -> Arc<Task> {
+fn create_user_task(user_space: Arc<VmSpace>, user_ctx: Arc<UserContext>) -> Arc<Task> {
```

Comment:
This `user_space` variable should be renamed to `vm_space` in order to avoid confusion.